src/GitHash.hs

```diff
@@ -147,13 +144,32 @@ getGitInfo root = try $ do
   packedExists <- doesFileExist packedRefs
   let files3 = if packedExists then [packedRefs] else []
 
-      _giFiles = concat [files1, files2, files3]
-      run args = do
+  return $ concat [files1, files2, files3]
+
+-- | Get a list of dependent files from a @.git@ file.
+getGitFilesFromFile :: FilePath -> IO [FilePath]
```

Comment:
This function seems more confusing than helpful. The name doesn't seem to indicate at all what it's doing.